ui/app/controllers/clients/index.js

```diff
@@ -164,11 +169,35 @@ export default class IndexController extends Controller.extend(
     return volumes.sort().map((volume) => ({ key: volume, label: volume }));
   }
 
+  @computed('selectionNodePools', 'model.nodePools.[]')
+  get optionsNodePools() {
+    const availableNodePools = this.model.nodePools;
```

Comment:
One interesting bit here that I hadn't considered is the special `all` node pool. It's kind of like a "virtual" node pool, in the sense that it always includes all nodes, no matter what node pool they are actually part of.

So I think it would make to remove it from the `availableNodePools` so it's not a dropdown option. Otherwise we would try to match nodes with node pool set to `all`, but this is not allowed:
https://github.com/hashicorp/nomad/blob/84e7cf39f6c7b0aee49d174ba03bd95824c83c59/command/agent/command.go#L390-L393

Reply:
Fixed in https://github.com/hashicorp/nomad/pull/17633